src/data.py

```diff
@@ -0,0 +1,132 @@
+import os, csv
+from  src.Num import Num
+from  src.sym import Sym
+from  src.cols import Cols
+from  src.row import Row
+
+'''
+Class `Data` is a holder of `rows` and their summaries (in `cols`)
+'''
+
+class Data:
+    '''
+    Constuctor which initializes the attributes
+
+    Attributes:
+    @cols: column class single object....no list here unlike rows....see next line :)
+    @rows: storing Rows class objects
+
+    Arguments:
+    @src: input file name
+    '''
+    def __init__(self, the, src) -> None:
+        self.cols = None 
+        self.rows = []
+        self.the = the
+        
+        '''
+        Check if the recived file name (variable src) is a string. 
+        If yes, then run csv function on it and also send add function name 
+        If no, we get that someone tried being cheeky here. They gave us a list of rows
+        of the file directly and didn't trust us :(
+        We simply use the add function directly here instead of asking csv function
+        to read each row and then use it.
+        '''
+        
+        if type(src) == str:
+            self.readFromCSV(src, self.add)
+        else:
+            for i in range(len(src)):
+                self.add(src[i])
+
+
+    
+    def readFromCSV(self, fname: str, funcnm) -> None:
+        """
+        Read content from CSV file and run a custom function on each row
+        @fname: file path with file name
+        @func: custom user function
+        """
+        sep = self.the['Seperator']
+        currentWorkingPath = os.path.dirname(__file__)
+        relativePath = os.path.join(currentWorkingPath, fname)
+        with open(relativePath, 'r') as file:
+            reader = csv.reader(file, delimiter=sep)
+            n = 0
+            for row in reader:
+                n = n + 1
+                t ={n : row}
+                funcnm(t)
+
+    
+    def add(self, xs):
+        if self.cols is None:
+            self.cols = Cols(list(xs.values())[0])
+        else:
+            temp_row = Row(xs)
+            self.rows.append(temp_row)
+            todo = self.cols.x + self.cols.y
+            # todo is a list of Sym and Num objects
```

Comment:
todo is a list of dependent and independent variables, it's definition would not exactly be Sym and Num objects, please change it to avoid confusion.

Reply:
Acknowledged.